### DISCUSSION TITLE
A way to make the library more robust against selector changes by google

### DISCUSSION THREAD

**Discussion by download13** [LuanRT/google-this#69] [Ideas]
It seems like keeping up with selector changes is a big issue for the reliability and maintenance burden of this library (#66 for example).
I wonder if there's a more flexible way of selecting the DOM subtrees that contain the information we want.

I did some searching on the subject and found [a paper](https://www.academia.edu/7289924/A_general_method_for_tree_comparison_based_on_subtree_similarity_and_its_use_in_a_taxonomic_database) that discusses ways of finding and matching subtrees within a larger tree structure.

Even a rigid, exact structural match would be more reliable than selectors, since DOM structure generally changes less often than randomly generated class names.

A fuzzy search that is able to tolerate small differences would be still more robust and likely only need modification if google were to do a complete overhaul of their results page structure.

I'm going to read through any papers I can find and see if there's a good way to do this in the context of a DOM tree, then maybe make a pull request. Just thought I'd leave this here in case anyone else had similar thoughts or wants to take over in the quite likely event that I forget about this tomorrow...

**Comment by LuanRT** [comment]
And just to add to this, we currently use the mobile & desktop version of Google. I think it might be a good idea to use the TV version instead as it rarely ever changes and doesn't use JavaScript to dynamically load content (plus the structure of the page is very very simple compared to the desktop and mobile pages).

**Reply by download13** [reply]
I didn't know there was a TV version. Can you post the endpoint url/pattern so I can mess with it a bit?

**Reply by LuanRT** [reply]
It's just a simpler version of the page for devices with less capabilities. Change your browser's user agent to `Mozilla/5.0 (ChromiumStylePlatform) Cobalt/Version`.

**Reply by RC047** [reply]
> It's just a simpler version of the page for devices with less capabilities. Change your browser's user agent to `Mozilla/5.0 (ChromiumStylePlatform) Cobalt/Version`.

it works for me at the moment, are there any other plans?

**Reply by LuanRT** [reply]
> > It's just a simpler version of the page for devices with less capabilities. Change your browser's user agent to `Mozilla/5.0 (ChromiumStylePlatform) Cobalt/Version`.
> 
> it works for me at the moment, are there any other plans?

Not for now, no. 

I haven't used this library in years, so it simply went unmaintained.
I may bring it back from the dead one day, but scraping Google now is just way too complicated with all the captchas and selector changes, so it'd probably just be a nightmare to maintain anyway.
